app.ts

```diff
@@ -0,0 +1,216 @@
+                startIndex = recordCount - tableRecordCount;
+            }
+            getRecords(startIndex, startIndex + end);
+        }
+    };
+
+    prevbtn.onclick = function() {
+        if (startIndex >= tableRecordCount) {
+            startIndex -= tableRecordCount;
+        } else if (startIndex < tableRecordCount) {
+            startIndex = 0;
+        }
+        getRecords(startIndex, startIndex + tableRecordCount - 1);
+    };
+
+    $("#canvas").append(fromField,searchbtn, prevbtn, nextbtn);
+
+    initiate();
+
+    /**
+     * Initiates the HTTP requests to obtain the records and column values necessary for the table.
+     */
+    function initiate() {
+        // get number of records:
+        $.ajax({
+            url: "http://localhost:2050/recordCount",
+            success: function (result) {
+                recordCount = parseInt(JSON.parse(result), 10);
+                // get columns:
+                $.ajax({
+                    url: "http://localhost:2050/columns",
+                    success: function (result) {
+                        columns = JSON.parse(result);
+                        numColumns = getSize(columns);
+                        // get first page of records and display them:
+                        adjustTableRecordCount();
+                        getRecords(0, tableRecordCount - 1);
+                    },
+                    error: function (err) {
+                        $("body").text("Error: " + err.status + " " + err.statusText);
+                    }
+                });
+            },
+            error: function (err) {
+                $("body").text("Error: " + err.status + " " + err.statusText);
+            }
+        });
+    }
+
+    /**
+     * Generates a table populated with data, distrubuted evenly.
+     * @param data The data records retrieved from the server, structured as a 2D object array.
+     * @param columns An object array containing the column heading values.
+     */
+    function generateTable(data: any, columns: any) {
+        $("table").remove(); // Remove previous table
+        let dataSize = getSize(data);
+        let table = document.createElement("table");
+        let tbdy = document.createElement("tbody");
+        table.appendChild(generateHeadings(columns));
+        for (let i = 0; i < tableRecordCount; i++){
+            let tr = document.createElement("tr");
+            for (let j = 0; j < numColumns; j++){
+                let td = document.createElement("td");
+                if (dataSize > i)
+                    td.appendChild(document.createTextNode(data[i][j]));
+                tr.appendChild(td);
+            }
+            tbdy.appendChild(tr);
+        }
+        table.appendChild(tbdy);
+        canvas.appendChild(table);
+    }
+
+    /**
+     * Finds the number of entries within a data set.
+     * @param data An array of data, type is unknown (any).
+     */
+    function getSize(data: any) {
```

Comment:
Your method signature seems to return a number, so I would advise that you append the return type to the method signature and leverage the type safety that Typescript makes available.
```ts
    function getSize(data: any): number {
        // Implementation
    }
```
Also, it's not great practice to make your function accept any values, because then we are not leveraging the type safety that TS makes available to us.